### ISSUE TITLE
Desktop width and layout : boxed option ?

### ISSUE THREAD

**Issue by ghost** [Nikeo/customizr#1265]
- How complex would it be to propose a boxed layout <=> wrapping header, main content and footer in a predefined width ( default would be 1140px ) ?

**Comment by eri-trabiccolo** [comment]
Hi

- would rather go with a solution that treats the blocks in a separate way. E.g. This will allow us to add another option to 'contain' the header only. It requires some html changes, but shouldn't be that hard. I guess the slider full in this case will be forced to boxed...
- for that min width only? Shouldn't be a big big issue but that has a slight impact on the image sizes too.
At least. 
Just rapid thoughts, need more in depth analysis.

**Comment by Nikeo** [comment]
@media (min-width: 1200px)
style.css?ver=4…1509976964:460
.container {
    width: 1140px;
    max-width: 100%;
}
@media (min-width: 992px)
style.css?ver=4…1509976798:453
.container {
    width: 960px;
    max-width: 100%;
}

@media (min-width: 768px)
style.css?ver=4…1509976964:446
.container {
    width: 720px;
    max-width: 100%;
}

@media (min-width: 576px)
style.css?ver=4…1509976964:439
.container {
    width: 540px;
    max-width: 100%;
}

**Comment by ghost** [comment]
# Spec

Simple boxed option on page wrap, like

```
#tc-page-wrap {
    position: relative;
    background: inherit;
    z-index: 1;
    width: 1140px;
    margin: auto;
}
```

## Issues

- sticky header to adapt

**Comment by eri-trabiccolo** [comment]
(make a first a prototype)

**Comment by eri-trabiccolo** [comment]
first prototype branch can be found here:
https://github.com/eri-trabiccolo/customizr/tree/boxed-layout
(css and php files are not concatenated there)

**Comment by ghost** [comment]
OK cool.
Can you create a beta PR with it ?

**Comment by ghost** [comment]
## padding
- .container selector should have a padding of 30px left and right , instead 15px currently => potential side effects ?

## customizer options
There should one main setting, let's say "Site layout", a select input with 2 options : Wide (default) and Boxed
When Wide is selected, two selectsub-settings are available
- Tobpar layout
- Main Header layout
- Footer Credits layout

**Comment by eri-trabiccolo** [comment]
Another PR pushed (https://github.com/presscustomizr/customizr/pull/1323)
Things you might want to verify in boxed mode:
- ~~when main header (and or site layout) section is boxed: mobile dropdown search and collapsing menu width when ww > 565px (tablets). To me it's fine that it doesn't take the whole header width but the "contained" one~~:
<img width="582" alt="example" src="https://user-images.githubusercontent.com/7689242/33323918-410c948a-d44e-11e7-8390-6540f67eafe5.png">

~~The alternative would be to have the "background" edge-to-egde, but the inner elements (Search label, and input) still "boxed": it would require some additional work though.~~
_The **alternative** explained above has been implemented in https://github.com/presscustomizr/customizr/pull/1323/commits/2a98efd0241817a64b5dac20b6bab8ee6cf0c869_


- In order to keep inner elements at the same width, .container selector not only will have a padding of 30px left and right, but its width will be increased of 30px too. This means the the "box" (in desktops) width will be 1170px, but the grid-container (for example) width, will still be 1110px as in the "wide" layout.
- Do we want to add an additional slider-full image size when using the boxed layout? The current slider-full image size has "unlimited" width (9999), while in a boxed layout it could at maximum have a width of 1170px (this is a direct consequence of what I decided regarding to the box width: see the above bullet).
